editions/tw5.com/tiddlers/widgets/ListWidget.tid

```diff
@@ -117,10 +119,29 @@ Displays as:
 </$list>
 <<<
 
-Note that using the `counter` attribute can reduce performance when working with list items that dynamically reorder or update themselves. The best advice is only to use it when it is really necessary: to obtain a numeric index, or to detect the first or last entries in the list.
+Note that using the `counter` attribute can reduce performance when working with list items that dynamically reorder or update themselves. The best advice is only to use it when it is really necessary: to obtain a numeric index, or to detect the first or last entries in the list. Note that if you are only using it to insert something (like a comma) between list items, the `join` or `joinText` attributes perform much better and you should use them instead of `counter`.
 
 Setting `counter="transclusion"` is a handy way to make child elements for each list element be identified as unique. A common use case are multiple [[tag macros|tag Macro]] for the same tag generated by a list widget. Refer to [[tag macro examples|tag Macro (Examples)]] for more details.
 
+!! `join` and `joinText` attributes
+
+The optional `join` or `joinText` attributes allow you to insert something between each list item without needing to use the `counter` attribute, which can become quite slow if the list is updated frequently. The difference between them is that the `join` attribute will have its contents wikified, while `joinText` will not. If both `join` and `joinText` are given, `joinText` has precedence and `join` will be ignored.
+
+<<.tip """Wikifying the attribute value is a relatively slow process, so if all you need is a bit of plain text (say, a comma and a space) between list items, `joinText` will perform a lot better than `join`. But if you need some HTML element like `<br>` inserted between list items, then use the `join` attribute.""">>
```

Comment:
Did you think about to do a different nameing. eg: `join` is the text/plain parameter and `joinWikified` the text/vnd.tiddlywiki one?

Reply:
I did. In https://github.com/Jermolene/TiddlyWiki5/pull/7694/commits/71a4a9fc0acd791b0c01051ab21506c129e83c46 I had `join` be the plaintext paramenter, and `joinTemplate` (not a good name, `joinWikified` is better) be the wikified one. But then I thought that people will expect `join="<br/>"` to work and will be confused if it doesn't, so in the next commit (https://github.com/Jermolene/TiddlyWiki5/pull/7694/commits/d0c23ed5abd79c70151f7598b2dab3cfe501951b) I switched it back to `join` being the wikified one and `joinText` being the plain-text one.

I'm open to suggestions, though. I actually prefer `join` being plain text, but would need to hear opinions from users. Maybe I should start a thread on https://talk.tiddlywiki.org/.

Reply:
>Maybe I should start a thread on https://talk.tiddlywiki.org/.

I do not think so. It will be 50/50. IMO @Jermolene should decide it and we are done.